internals/docs/guides/inspections.md

```diff
@@ -1,3 +1,114 @@
+
+```c#
+void callback(object sender, EventReceivedEventArgs e)
+{
+    ...
+}
+
+void getInspectableDataFromEvents()
+{
+    Inspector.EventReceived += callback;
+}
+```
+
+When a new event arrives, callback is called. To read the inspection data out of the inspection context use `GetInspectableData()`:
+
+```c#
+void callback(object sender, EventReceivedEventArgs e)
+{
+    if (e.Context.ModuleID == "org.tizen.system.crash")
+    {
+        string[] parameters = {"cs_info_json"};
+        InspectionData data = e.Context.GetInspectableData(parameters);
+    }
+
+    ...
+}
+```
+
+Use the `parameters` argument to specify which data to read from the inspection context. Available parameters depend on the module that provides the context.
+
+In case of `"org.tizen.system.crash"` module, the proper values are:
+
+-   `"cs_info_json"`: JSON report is returned, which contains information about crashed process such as callstack, mapped memory regions, and so on.
+-   `"cs_full"`: Full crash report ZIP archive is collected.
+
+The obtained data is associated with the `data` Inspections handler. To read the data content, use `Read()` method:
+
+```c#
+void callback(object sender, EventReceivedEventArgs e)
+{
+    if (e.Context.ModuleID == "org.tizen.system.crash")
+    {
+        string[] parameters = {"cs_info_json"};
+        InspectionData data = e.Context.GetInspectableData(parameters);
+
+        byte[] buf = new byte[100];
+        while (true)
+        {
+            // Read data and check for EOF
+            if (data.Read(buf, 0, buf.Length) == 0)
+                break;
+
+            // Log received data
+            Log.Info("InspectionsExample", System.Text.Encoding.UTF8.GetString(buf));
+        }
+    }
+}
+```
+
+## Request inspection data
+
+`Tizen.Inspections` can also request data from other application or service, especially the inspection data that for various reasons cannot or must not be saved in the logs, such as binary or in custom format data, large volume data, reports that are time-consuming to create, and so on.
+
+To do so, call `RequestInspectableData()`. The `moduleId` parameter must be the id of the process that supports `Tizen.Inspections` requests:
+
+```c#
+
+void requestInspectableData()
+{
+    string moduleId = "org.tizen.some_service";
+    string[] parameters = {"custom_parameter"};
+
+    InspectionData data = Inspector.RequestInspectableData(moduleId, parameters);
+
+    byte[] buf = new byte[100];
+    while (true)
+    {
+        // Read data and check for EOF
+        if (data.Read(buf, 0, buf.Length) == 0)
+            break;
+
+        // Log received data
+        Log.Info("InspectionsExample", System.Text.Encoding.UTF8.GetString(buf));
+    }
+}
+```
+
+As in above example, after successful request, data may be read with `Read()` method.
+
+## Related information
+- Dependencies
+  - Tizen 6.0 and Higher for Mobile
+  - Tizen 6.0 and Higher for Wearable
```

Comment:
How about removing "for Mobile" and "for Wearable"?

Reply:
Done.